token/cli/src/main.rs

```diff
@@ -8456,6 +9003,87 @@ mod tests {
             .unwrap();
         assert!(bool::from(extension.allow_non_confidential_credits));
 
+        // deposit confidential tokens
+        let deposit_amount = 100.0;
+        mint_tokens(&config, &payer, token_pubkey, deposit_amount, token_account).await;
+
+        process_test_command(
+            &config,
+            &payer,
+            &[
+                "spl-token",
+                CommandName::DepositConfidentialTokens.into(),
+                &token_pubkey.to_string(),
+                &deposit_amount.to_string(),
+            ],
+        )
+        .await
+        .unwrap();
+
+        // apply pending balance
+        process_test_command(
+            &config,
+            &payer,
+            &[
+                "spl-token",
+                CommandName::ApplyPendingBalance.into(),
+                &token_pubkey.to_string(),
+            ],
+        )
+        .await
+        .unwrap();
+
+        // confidential transfer
+        let destination_account = create_auxiliary_account(&config, &payer, token_pubkey).await;
+        process_test_command(
+            &config,
+            &payer,
+            &[
+                "spl-token",
+                CommandName::ConfigureConfidentialTransferAccount.into(),
+                "--address",
+                &destination_account.to_string(),
+            ],
+        )
+        .await
+        .unwrap(); // configure destination account for confidential transfers first
+
+        // NOTE: the test fails due to transaction size limit
+        // let transfer_amount = 100.0;
+        // process_test_command(
+        //     &config,
+        //     &payer,
+        //     &[
+        //         "spl-token",
+        //         CommandName::Transfer.into(),
+        //         &token_pubkey.to_string(),
+        //         &transfer_amount.to_string(),
+        //         &destination_account.to_string(),
+        //         "--confidential",
+        //     ],
+        // )
+        // .await
+        // .unwrap();
+
+        // withdraw confidential tokens
+        //
+        // NOTE: the test fails due to transaction size limit :(
```

Comment:
We'll just need another helper on the token client to generate the proof here, right?

Reply:
Yeah, it should suffice to build a token-client function that (1) creates a withdraw proof context, and then (2) follow through with the withdraw instruction.